controllers/vspherevm_controller.go

```diff
@@ -192,13 +189,15 @@ func (r vmReconciler) Reconcile(ctx context.Context, req ctrl.Request) (_ ctrl.R
 	// in that case nil vsphereMachine can cause panic and CrashLoopBackOff the pod
 	// preventing vspheremachine_controller from setting the ownerref
 	if err != nil || vsphereMachine == nil {
-		r.Logger.Info("Owner VSphereMachine not found, won't reconcile", "key", req.NamespacedName)
+		log.Info("Owner VSphereMachine not found, won't reconcile", "key", req.NamespacedName)
 		return reconcile.Result{}, nil
 	}
 
+	log = log.WithValues("VSphereVM", klog.KObj(vsphereMachine))
+	ctx = ctrl.LoggerInto(ctx, log)
 	vsphereCluster, err := util.GetVSphereClusterFromVSphereMachine(ctx, r.Client, vsphereMachine)
 	if err != nil || vsphereCluster == nil {
-		r.Logger.Info("VSphereCluster not found, won't reconcile", "key", ctrlclient.ObjectKeyFromObject(vsphereMachine))
+		log.Info("VSphereCluster not found, won't reconcile", "key", ctrlclient.ObjectKeyFromObject(vsphereMachine))
 		return reconcile.Result{}, nil
 	}
 
```

Comment:
Let's add the VSphereCluster to the logger here (+ update the ctx with the new logger)